### PR TITLE
Data stats [Delivers #147284685]

### PR DESCRIPTION
**Pull request by barrywark**
Returns organization/project file usage stats:

```
{:stats [{:id                        project-uuid
              :organization_id  Int
              :usage                 Num}]}
```

For org 0, returns usage for projects _owned_ by current user.

For private orgs, returns usage for all org projects.

Depends on `revision-bytes` view:

```
function (doc) {
    var org = doc.organization;

    if(doc.type && doc.type === 'Revision') {
      if(!doc.trash_info) {
        if(doc.attributes && doc.attributes.content_length) {
            emit([org,doc.owner,doc._id], doc.attributes.content_length);

            if (doc.links && doc.links._collaboration_roots) {
                var roots = doc.links._collaboration_roots;
                for (var i = 0; i < roots.length; i++) {
                    emit([org, roots[i], doc.owner, doc._id], doc.attributes.content_length);
                }
            }
        }
      }
    }
}
```

### CODE REVIEW

**Comment by dguettler** [comment]
@barrywark yes I think this looks correct. Thank you.

**Comment by dguettler** [comment]
@barrywark actually one missing part
both organization and project should have an entry in the `links` section to `stats` - don't know if this is currently added by the webapi or just passed through from Rails

File: src/ovation/storage.clj [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  "Gets the provided organization file storage (usage) in bytes.
+  For organization 0, returns usage for all Projects belonging to the current user.
+
+  Conveys a seq of {:project_id, :org_id, :usage}"
+  [ctx db org-id ch & {:keys [close?] :or {close? true}}]
+
+  (let [org0?       (= 0 org-id)
+        user-id     (request-context/user-id ctx)
+        startkey    (if org0? [user-id] [org-id])
+        endkey      (if org0? [user-id {}] [org-id, {}])
```
Comment (dguettler) [inline src/ovation/storage.clj]: Is this correct here? I'd assume those 2 statements to look the same?

Comment (barrywark) [inline src/ovation/storage.clj]: @dguettler comma is optional in Clojure, so `[org-id, {}]` and `[org-id {}]` are the same. It's bad style to have the comma, however.

**Pull request merged**